servicex/cache.py

```diff
@@ -56,6 +60,30 @@ def set_query(self, json: Dict[str, str], v: str):
         with f.open('w') as o:
             o.write(f'{v}\n')
 
+    def set_query_status(self, query_info: Dict[str, str]):
+        '''Cache a query status (json dict)
+
+        Args:
+            query_info (Dict[str, str]): The info we should cache. Must contain `request_id`.
+        '''
+        assert 'request_id' in query_info, \
+            "Internal error - request_id should always be part of info returned"
```

Comment:
Assertions [should not be used for error handling](https://stackoverflow.com/questions/20076182/what-are-the-implications-of-using-assert-vs-raise-exception). Instead, we should raise a custom exception.

Reply:
I've been using them like this:

- These are things that should never happen, but if they do, they will be hard to figure out why they happened from the user end.
- They require a whole extra test to provide code-coverage if I use an if statement.

I could just remove it: after all, we never expect this.

Reply:
Okay, that's a perfectly good way to use them (I wasn't sure how rare this error was). It's what the article I linked describes as a sanity check. For example, if you had a function `square = lambda x: x*x` then you might do:
```
y = square(x)
assert y >= 0
```
which should never execute if x is a real number.